py_mmd_tools/xml_util.py

```diff
@@ -0,0 +1,58 @@
+"""
+Utility tool to work on mmd xml files.
+ License:
+     This file is part of the S-ENDA-Prototype repository (https://github.com/metno/py-mmd-tools).
+     S-ENDA-Prototype is licensed under GPL-3.0 (https://github.com/metno/py-mmd-tools/blob/master/LICENSE)
+"""
+
+import pathlib
+import time
+import errno
+import os
+import lxml.etree as ET
+from datetime import datetime
+from lxml.etree import XMLSyntaxError
+
+
+def xml_check(xml_file):
+    """[validate xml syntax from filepath]
+    Args:
+        xml_file ([str]): [filepath to an xml file]
+    Returns:
+        [bool]: [return True if a valid xml filepath is provided, 
+        raises an exception if the xmlfile is invalid, empty, or doesn't exist ]
+    """
+    if pathlib.Path(xml_file).is_file() and os.path.getsize(xml_file) != 0:
+        try:
+            xml = ET.parse(xml_file)
```

Comment:
isn't it enough with this line? FileNotFound and XMLSyntaxError should be raised by `ET.parse`, shouldn't it?

Reply:
I may be wrong, but it looks like `ET.parse` raises a `OSerror`  when a non existent file is given directly, example below.:

```python
In [1]: import lxml.etree as ET  
In [2]: xml = ET.parse('tests/data/this_fine_not_exist.xml')                                                                                                                  
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
<ipython-input-7-60df23105c82> in <module>
----> 1 xml = ET.parse('tests/data/this_fine_not_exist.xml')

src/lxml/etree.pyx in lxml.etree.parse()

src/lxml/parser.pxi in lxml.etree._parseDocument()

src/lxml/parser.pxi in lxml.etree._parseDocumentFromURL()

src/lxml/parser.pxi in lxml.etree._parseDocFromFile()

src/lxml/parser.pxi in lxml.etree._BaseParser._parseDocFromFile()

src/lxml/parser.pxi in lxml.etree._ParserContext._handleParseResultDoc()

src/lxml/parser.pxi in lxml.etree._handleParseResult()

src/lxml/parser.pxi in lxml.etree._raiseParseError()

OSError: Error reading file 'tests/data/not_a_valid_xmlh.xml': failed to load external entity "tests/data/this_fine_not_exist.xml"
```